src/viperleed/calc/classes/state_recorder.py

```diff
@@ -0,0 +1,49 @@
+"""Module state_recorder of viperleed.calc.classes.
+
+Defines a class to record and retrieve calculation states.
+"""
+
+__authors__ = (
+    'Alexander M. Imre (@amimre)',
+    )
+__copyright__ = 'Copyright (c) 2019-2024 ViPErLEED developers'
+__created__ = '2024-08-06'
+__license__ = 'GPLv3+'
+
+from collections import namedtuple
+from copy import deepcopy
+
+from viperleed.calc.sections.calc_section import CalcSection
+
+# A state is given by a tuple (slab, rparams).
+State = namedtuple('State', 'slab, rparams, section')
```

Comment:
Perhaps we want to have the `'rparams'` attribute named `'rpars'`, to better keep consistency with other spots, and not get confused with the module name. the same is true (and especially important) for the argument name in the `StateRecorder` methods.